miner/worker_test.go

```diff
@@ -47,73 +47,51 @@ const (
 	testGas = 144109
 )
 
-type testCase struct {
-	testTxPoolConfig  core.TxPoolConfig
-	ethashChainConfig *params.ChainConfig
+var (
+	// Test chain configurations
+	testTxPoolConfig      core.TxPoolConfig
+	ethashChainConfig     *params.ChainConfig
+	tendermintChainConfig *params.ChainConfig
 
-	testBankKey     *ecdsa.PrivateKey
-	testBankAddress common.Address
-	testBankFunds   *big.Int
+	// Test accounts
+	testBankKey, _  = crypto.GenerateKey()
+	testBankAddress = crypto.PubkeyToAddress(testBankKey.PublicKey)
+	testBankFunds   = big.NewInt(1000000000000000000)
 
-	testUserKey     *ecdsa.PrivateKey
-	testUserAddress common.Address
-	testUserFunds   *big.Int
+	testUserKey, _  = crypto.GenerateKey()
+	testUserAddress = crypto.PubkeyToAddress(testUserKey.PublicKey)
 
 	// Test transactions
 	pendingTxs []*types.Transaction
 	newTxs     []*types.Transaction
 
-	testConfig *Config
-}
-
-func getTestCase() (*testCase, error) {
-	testBankKey, err := crypto.GenerateKey()
-	if err != nil {
-		return nil, err
-	}
-
-	testUserKey, err := crypto.GenerateKey()
-	if err != nil {
-		return nil, err
-	}
-
-	t := &testCase{
-		testBankKey:     testBankKey,
-		testBankAddress: crypto.PubkeyToAddress(testBankKey.PublicKey),
-		testBankFunds:   big.NewInt(1000000000000000000),
-
-		testUserKey:     testUserKey,
-		testUserAddress: crypto.PubkeyToAddress(testUserKey.PublicKey),
-		testUserFunds:   big.NewInt(1000),
-
-		testConfig: &Config{
-			Recommit: time.Second,
-			GasFloor: params.GenesisGasLimit,
-			GasCeil:  params.GenesisGasLimit,
-		},
-	}
-
-	t.testTxPoolConfig = core.DefaultTxPoolConfig
-	t.testTxPoolConfig.Journal = ""
-
-	t.ethashChainConfig = &params.ChainConfig{}
-	*t.ethashChainConfig = *params.TestChainConfig
-
-	tx1, err := types.SignTx(types.NewTransaction(0, t.testUserAddress, t.testUserFunds, params.TxGas, nil, nil), types.HomesteadSigner{}, testBankKey)
-	if err != nil {
-		return nil, err
-	}
-	t.pendingTxs = append(t.pendingTxs, tx1)
-
-	tx2, err := types.SignTx(types.NewTransaction(1, t.testUserAddress, t.testUserFunds, params.TxGas, nil, nil), types.HomesteadSigner{}, testBankKey)
-	if err != nil {
-		return nil, err
+	testConfig = &Config{
+		Recommit: time.Second,
+		GasFloor: params.GenesisGasLimit,
+		GasCeil:  params.GenesisGasLimit,
 	}
-	t.newTxs = append(t.newTxs, tx2)
+)
 
+func init() {
+	testTxPoolConfig = core.DefaultTxPoolConfig
+	testTxPoolConfig.Journal = ""
+	ethashChainConfig = params.TestChainConfig
+	tendermintChainConfig = &params.ChainConfig{}
+	tendermintChainConfig = params.AutonityTestChainConfig
```

Comment:
Can you please explain why is `tendermintChainConfig` set twice? I think the following is sufficient

```suggestion
	tendermintChainConfig = params.AutonityTestChainConfig
```

I ran the worker_test.go file with the above suggestion, it passes.

Reply:
great thanks !